.github/workflows/perf.yml

```diff
@@ -0,0 +1,32 @@
+name: Build performance
+
+on:
+  push:
+    # branches:
+    #   - master
+  pull_request:
+    branches:
+      - master
+      - 'releases/**'
+
+
+jobs:
+  build_performance:
+    runs-on: ubuntu-latest
+    container: gitlab-registry.cern.ch/acts/machines/ubuntu2004:v2
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: pip3 install git+https://github.com/paulgessinger/cmakeperf.git@2a110c1
+      - name: Configure
+        run: >
+          cmake -B build -S .
+          -GNinja
+          -DCMAKE_BUILD_TYPE=RelWithDebInfo
+          -DCMAKE_CXX_FLAGS="-Werror"
+          -DACTS_BUILD_EVERYTHING=ON
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+      - name: Measure
+        run: cmakeperf collect build/compile_commands.json -o perf.csv -j$(nproc)
```

Comment:
If you want stable time measurements, parallelism is dangerous because you don't know what's running in parallel with you and it may interact badly through some shared resources (memory bus, storage...).

Personally, I care most about RAM consumption right now, and think that timing measurements on cloud VMs are a lost cause, so I do not particularly care.

Reply:
Right. I wouldn't give too much weight to the time measurements anyway. It speeds up the job a little bit, and the memory measurement should be robust enough with concurrency.